packages/kit/src/exports/vite/dev/index.js

```diff
@@ -30,10 +30,11 @@ export async function dev(vite, vite_config, svelte_config) {
 	}
 
 	const fetch = globalThis.fetch;
-	globalThis.fetch = (info, init) => {
+	globalThis.fetch = async (info, init) => {
 		if (typeof info === 'string' && !/^\w+:\/\//.test(info)) {
+			await Promise.resolve(); // allows people to use malformed fetch urls in `{#await ..}` (native fetch error is swallowed by those)
```

Comment:
This came out of https://github.com/sveltejs/kit/issues/8536#issuecomment-1385355799 , doing `{#await fetch('/foo')}..` worked previously, because the error was swallowed.

Either we treat this as a potential breaking change because this worked previously; also it may be deliberate to rely on this behavior on the server to save some code ("yeah this will fail but I don't care, it will work on the client then"). Or we treat it as a bug that this ever worked and revert the 2nd commit of this PR.

Reply:
I'm in favour of reverting — this behaviour seems much too magical to me

Reply:
What's magical about it for you? That this relies on specific behavior of `#await`? I'm not sure if that's the case but maybe this also works if we omit that line if the function is marked as async? Would that be too magical, too?
Do you think it's edge case enough that we can mark this as "this merely uncovered a bug in your code and is not a breaking change"?

Reply:
Rule 1: don't swallow errors.

If my intent is to have an `await` that only works in the browser, then that's the code I should write. It's far more likely that someone simply didn't understand the mechanics of `await` and `fetch`, and this kind of benevolent interference prevents them from developing that understanding (while causing confusion to someone who understands why it _shouldn't_ work).

This definitely qualifies as 'we found a bug in your code', yeah